### ISSUE TITLE
🔧 Fix Lab Mode Training Pipeline & Complete ML/RL Infrastructure

### ISSUE THREAD

**Issue by kevinsuero072897-collab** [Quotraders/QBot#624]
# 🎯 Mission: Fix Training Pipeline & Complete ML/RL Architecture

### 1. Training Freeze - Old DLL Cache Issue
**Problem**: Lab Mode training freezes at data loading step. Mystery log message "Fetching historical data using Python script..." appears despite being deleted from source code.

**Root Cause**: Old compiled DLL being cached somewhere (not in bin/obj folders - already deleted multiple times)

**Evidence**:
- Source code: `HistoricalDataBridgeService.cs` has Lab Mode detection (lines 97-107, 158-165, 248-256)
- Runtime logs: Old log message still appearing at [15:59:26.579], [16:04:08.621]
- grep confirms: Log message does NOT exist in any .cs file
- Multiple clean/rebuild cycles have NOT fixed it

**Fix Required**:
1. Nuclear option: Delete global NuGet package cache (`~\.nuget\packages`)
2. Force restore: `dotnet restore --force`
3. Complete rebuild: `dotnet clean && dotnet build`
4. Verify runtime logs match source code

### 2. CVaR-PPO Champion Pointer Missing
**Problem**: `CVaR-PPO_champion.txt` file missing, causing bootstrap to re-run every startup

**Evidence**:
- Model metadata exists: `CVaR-PPO_v1.0.0-bootstrap.json` ✅
- Champion pointer missing: `CVaR-PPO_champion.txt` ❌
- Other 8 champions have pointers ✅

**Fix Required**:
Create `src/UnifiedOrchestrator/model_registry/CVaR-PPO_champion.txt` with content: `v1.0.0-bootstrap`

---

### 3. Missing ML Component Trainers
**Problem**: Bootstrap registers 9 champions but only 2 have actual trainers implemented

**Implemented Trainers** (2/9):
- ✅ CVaR-PPO (`CVaRPPOTrainer.cs`)
- ✅ Neural-UCB (`NeuralUcbBanditTrainer.cs`)

**Missing Trainers** (7/9):
- ❌ LSTM (referenced in `InferenceBrain.cs` line 206, but no trainer exists)
- ❌ Regime-Detector (bootstrap placeholder only)
- ❌ Model-Ensemble (bootstrap placeholder only)
- ❌ Pattern-Recognition (bootstrap placeholder only)
- ❌ Slippage-Latency-Model (bootstrap placeholder only)
- ❌ S15-RL-Policy (validated but not trained)
- ❌ PM-Optimizer (optimization only, no trainer)

**Training Pipeline** (`HistoricalTrainingOrchestrator.cs` lines 968-979):
1. `TrainCVarPPOAsync()` - ✅ Implemented
2. `TrainNeuralUCBAsync()` - ✅ Implemented
3. `TrainLSTMAsync()` - ⚠️ Stub/placeholder
4. `OptimizePositionManagementAsync()` - ⚠️ Optimization only
5. `RunS15ShadowValidationAsync()` - ⚠️ Validation only

**Decision Required**:
- Option A: Implement missing trainers (LSTM, Pattern-Recognition, etc.)
- Option B: Mark as "integrated" components and update bootstrap to only register CVaR-PPO + Neural-UCB as trainable
- Option C: Keep bootstrap as-is but document which are placeholders vs trainable

### 4. No Trained ONNX Models Exist
**Problem**: Zero `.onnx` files in artifacts, all champions stuck on bootstrap placeholders

**Expected**:
```
artifacts/models/cvar_ppo_v1.1.0.onnx
artifacts/models/neural_ucb_v1.1.0.onnx
artifacts/models/lstm_v1.0.0.onnx
```

**Found**: ZERO .onnx files

**Impact**: Trading system running with no actual trained ML models!

**Fix**: Complete one successful training run to generate first ONNX artifacts

---

## 📋 VERIFICATION CHECKLIST

After fixes, verify:
- [ ] Lab Mode training completes without freezing
- [ ] CVaR-PPO and Neural-UCB produce ONNX files
- [ ] Model registry promotions work (challengers → champions)
- [ ] Bootstrap stops re-running on every startup
- [ ] Training logs match source code (no mystery messages)
- [ ] All 9 champion pointers exist
- [ ] ONNX models load in UnifiedTradingBrain
- [ ] Experience repository collecting data in Terminal Mode

---

## 🎯 SUCCESS CRITERIA

**Phase 1 - Training Works**:
- Training completes Sunday session without freezing
- At least 2 ONNX models generated (CVaR-PPO, Neural-UCB)
- Champion promotion successful (v1.0.0-bootstrap → v1.1.0-trained)

**Phase 2 - Complete Architecture**:
- All 9 champion pointers exist
- Missing trainers implemented or documented as integrated
- Architecture diagram updated to reflect actual vs placeholder components

**Phase 3 - Production Ready**:
- End-to-end test: Terminal Mode → Lab Mode → Promotion → Terminal Mode
- Verify trained models actually improve trading performance
- Confirm no API calls in Lab Mode (segregation maintained)

---

## 📁 KEY FILES TO MODIFY

1. **Fix Training Freeze**:
   - Check: `src/UnifiedOrchestrator/Services/HistoricalDataBridgeService.cs` (Lab Mode checks)
   - Check: `src/UnifiedOrchestrator/Services/HistoricalTrainingOrchestrator.cs` (training pipeline)
   - Nuclear: Delete NuGet cache, force rebuild

2. **Fix Champion Pointer**:
   - Create: `src/UnifiedOrchestrator/model_registry/CVaR-PPO_champion.txt`

3. **Complete Trainers** (if implementing):
   - Implement: `src/UnifiedOrchestrator/Services/LSTMTrainer.cs`
   - Implement: `src/UnifiedOrchestrator/Services/PatternRecognitionTrainer.cs`
   - Implement: `src/UnifiedOrchestrator/Services/SlippageLatencyTrainer.cs`

4. **Update Bootstrap**:
   - Modify: `src/UnifiedOrchestrator/Services/ModelRegistryBootstrapService.cs` (document placeholders)

---

## 🔍 DEBUGGING CONTEXT

**Historical Data**: ✅ Present & Recent
- `ES_90days.json`: 704 KB (Oct 20, 2025)
- `NQ_90days.json`: 705 KB (Oct 20, 2025)

**Infrastructure**: ✅ 95% Complete
- TrainingOrchestrator, MetricsCollector, AlertService, RetryService, CheckpointService all exist

**Experience Collection**: ⚠️ Unknown
- ExperienceRepository exists, but need to verify Terminal Mode is collecting

**Python Bridge**: ⚠️ Minimal
- `sdk_bridge.py` exists, but limited training scripts

---

## 💬 ADDITIONAL NOTES

User's gut feeling was correct - infrastructure is enterprise-grade but no trained models exist. Like having a Formula 1 pit crew with no race car!

Priority: **Fix training freeze FIRST**, then worry about missing trainers. Can't complete architecture until training actually works.

Current state: Sunday training enabled (hardcoded `return true;` in scheduler per user request), waiting for successful run.
